torch/utils/cpp_extension.py

```diff
@@ -153,6 +153,15 @@ def _join_rocm_home(*paths) -> str:
 
                               !! WARNING !!
 '''
+CUDA_MISMATCH_MESSAGE = '''
+The detected CUDA version ({0}) mismatches the version that was used to compile
```

Comment:
nit: consider using named interpolates rather than positional. Also, I know the existing code has the warning out of line, but since we've got f-strings now, consider putting the strings directly inline in the site where they're used; I typically find it easier to understand the code when everything is in one place. (Assuming everything else in the PR is fine I plan to merge this as is.)